swarm/pss/pss_test.go

```diff
@@ -475,6 +482,196 @@ func testAsymSend(t *testing.T) {
 	}
 }
 
+// params in run name:
+// nodes/msgs/addrbytes/adaptertype
+// if adaptertype is exec uses execadapter, simadapter otherwise
+func TestNetwork(t *testing.T) {
+	t.Run("8/1024/2/sim", testNetwork)
+}
+
+func testNetwork(t *testing.T) {
+
+	type msgnotifyC struct {
+		id     discover.NodeID
+		msgIdx int
+	}
+	topic := whisper.BytesToTopic([]byte("foo:42"))
+	hextopic := common.ToHex(topic[:])
+
+	paramstring := strings.Split(t.Name(), "/")
+	nodecount, _ := strconv.ParseInt(paramstring[1], 10, 0)
+	msgcount, _ := strconv.ParseInt(paramstring[2], 10, 0)
+	addrsize, _ := strconv.ParseInt(paramstring[3], 10, 0)
```

Comment:
Rather than using a string to specify these, why not use an explicit, self-documenting type? For example:

```go
type testParams struct {
	NodeCount int
	MsgCount  int
	AddrSize  int
	Adapter   string
}

func (t *testParams) String() string {
	return fmt.Sprintf("%d/%d/%d/%s", t.NodeCount, t.MsgCount, t.AddrSize, t.Adapter)
}
```

Reply:
I'm not sure what you mean here. 

The idea is to have `TestNetwork` run `testNetwork` with different combinations of values, with the features that running [subtests][https://golang.org/pkg/testing/#T.Run] provides. The respective values are thus passed as strings.

How does using an intermediate struct contribute to more clarity here?